src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
+        this.first = null;
+        this.last = null;
+        this.size = 0;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (isEmpty()) {
+            first = last = newNode;
+        } else {
+            last.next = newNode;
+            newNode.prev = last;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexAdd(index);
+        Node<T> newNode = new Node<>(null, value, null);
+        if (isEmpty()) {
+            first = last = newNode;
+        } else if (index == 0) {
+            newNode.next = first;
+            first = newNode;
+        } else if (index == size) {
+            last.next = newNode;
+            newNode.prev = last;
+            last = newNode;
+        } else {
+            Node<T> current = findByIndex(index - 1);
+            newNode.next = current.next;
+            newNode.prev = current.prev;
+            current.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            Node<T> newNode = new Node<>(null,list.get(i), null);
+            add(newNode.item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> findNode = findByIndex(index);
+        return findNode.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> findNode = findByIndex(index);
+        T retunedValue = findNode.item;
+        findNode.item = value;
+        return retunedValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        return getRemovedItem(index);
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> removedNode = new Node<>(null, object, null);
+        int temp = 0;
+        for (Node<T> i = first; i != null; i = i.next) {
+            temp++;
+            if (removedNode.item == i.item || removedNode.item != null && removedNode.item
+                    .equals(i.item)) {
+                getRemovedItem(temp - 1);
```

Comment:
```suggestion
                unlink(temp);
```
why -1

Reply:
Index variable temp don't have value 0, that's why i need -1.